pkg/tfgen/docs.go

```diff
@@ -1357,129 +1361,139 @@ func (g *Generator) convertExamplesInner(
 		contract.IgnoreError(err)
 	}
 
-	for _, section := range splitGroupLines(docs, "## ") {
-		if len(section) == 0 {
-			continue
+	// Find code fences in the presented doc
+	type tfBlockIndex struct {
+		start       int
+		end         int
+		headerStart int // a value of -1 indicates there's no header
+	}
```

Comment:
A doc comment explaining exactly what these numbers mean would be great.

I'm assuming that `start` refers to the first "\`" in the "\`\`\`hcl" that opens the block, `end` refers to the last "\`" that closes the block (such that `text[end+1] == '\n'` in the common case, and `headerStart` refers to the opening "#" in the closest enclosing `## Some header` for the block.